src/api/spec/models/repository_spec.rb

```diff
@@ -135,4 +135,29 @@
       it { expect(subject.download_repositories.pluck(:arch, :url)).to contain_exactly([dod_repository.arch, dod_repository.url]) }
     end
   end
+
+  describe '#new_from_distribution' do
+    let(:target_project) { create(:project) }
+    let(:project) { create(:project_with_repository) }
+    let(:distribution) { create(:distribution, project: project, repository: project.repositories.first.name) }
+    let(:subject_repository) do
+      repository = Repository.new_from_distribution(distribution)
```

Comment:
is it possible to do it with Factories?

Reply:
Hm sure but what would we test then? `new_from_distribution` is the subject.